nixos/modules/services/blockchain/monero.nix

```diff
@@ -0,0 +1,232 @@
+{ config, options, pkgs, lib, ... }:
+
+with lib;
+
+let
+
+  description = "Monero Full Node";
+  walletDescription = "Monero Wallet RPC Daemon";
+
+  cu = pkgs.coreutils;
+
+  cfg = config.services.monerod;
+  opts = options.services.monerod;
+
+  configFile = optionalString (!isNull cfg.extraConfig) (" --config-file=${pkgs.writeText "monerod.conf" cfg.extraConfig}");
+
+  makeWalletJob = name: conf:
+  let
+
+    options = [
+      "--wallet-file=${conf.walletDir}/${conf.walletFile}"
+      (optionalString (!isNull conf.walletPassword) "--password=${conf.walletPassword}")
+      (optionalString (!isNull conf.walletPasswordFile) "--password-file=${conf.walletPasswordFile}")
+      (optionalString (!isNull conf.bindAddress) "--rpc-bind-ip=${conf.bindAddress}")
+      (optionalString (conf.confirmExternalBind) "--confirm-external-bind")
+      (optionalString (!isNull conf.bindPort) "--rpc-bind-port=${toString conf.bindPort}")
+      (optionalString (!isNull conf.login) "--rpc-login=${conf.login}")
+      ] ++ conf.extraOptions;
+
+  in
+
+  {
+
+    inherit (conf) enable;
+
+    description = "${walletDescription}: ${name}";
+    after = [ "network.target" ];
+    wantedBy = [ (if conf.independent then "multi-user.target" else "monerod.service") ];
+    bindsTo = [ (optionalString (!conf.independent) "monerod.service") ];
+
+    serviceConfig = {
+      User = cfg.user;
+      Group = cfg.group;
+      WorkingDirectory = cfg.dataDir;
+
+      ExecStartPre = [
+        "+${cu}/bin/chown -R ${cfg.user}:${cfg.group} ${conf.walletDir}"
+        "+${cu}/bin/chmod -R 700 ${conf.walletDir}"
+      ];
+      ExecStart = "${cfg.package}/bin/monero-wallet-rpc ${toString options}";
+
+      Restart = "always";
+      PrivateTmp = true;
+      TimeoutStopSec = 60;
+    };
+
+  };
+
+in
+
+{
+
+  options = {
+
+    services.monerod = {
+
+      enable = mkOption {
+        default = false;
+        description = "Enable ${description}";
+        type = types.bool;
+      };
+
+      package = mkOption {
+        default = pkgs.monero;
+        type = types.package;
+      };
+
+      user = mkOption {
+         default = "monero";
+         defaultText = "Monero daemon user"; 
+         type = types.str;
+      };
```

Comment:
since static user/group names are allocated, it is not really configurable. I would just drop these two options here and in every other service.